### DISCUSSION TITLE
Oqtane vs Radzen

### DISCUSSION THREAD

**Discussion by TrailBlazor** [oqtane/oqtane.framework#1463] [Q&A]
Hi, I'm a totally new to Oqtane. I'm currently using Radzen but I am somewhat limited with it. Has anyone switched from Radzen to Oqtane? Thanks, Dave.

**Comment by sbwalker** [comment]
Oqtane and Radzen are both accelerators for Blazor - but they solve different problems. Radzen provides a powerful suite of UI components. Oqtane provides a modular infrastructure for building full stack applications. You can use Radzen with Oqtane - please see the example Survey module https://github.com/oqtane/Oqtane.Survey @ADefWebserver

**Reply by TrailBlazor** [reply]
Thank you for the quick response. I'm mostly thinking of the framework and not the components. Radzen does not have 2FA and that concerns me. Also, your product is open source and that is not the case with Radzen Designer (their components are open source). So, I want to go to Oqtane but just seeing if anyone else has tried to do that and what pitfalls, if any.  Thanks

**Reply by sbwalker** [reply]
@TrailBlazor can you elaborate on what you mean by "Radzen does not have 2FA"?

**Reply by TrailBlazor** [reply]
I would like to be able to authenticate using google / facebook and maybe some other authenticators. Also when I think of two factor authenticating I think about sending a code to a mobile device and the user needs to enter that code. That would be just a one-time of forgot my password type situation.

**Reply by TrailBlazor** [reply]
*or forgot...

**Reply by sbwalker** [reply]
@TrailBlazor Oqtane does not currently support those authentication options either. It is on the roadmap for version 2.2 however the timing is difficult to judge because this is an open source project.

**Reply by TrailBlazor** [reply]
@sbwalker ah, okay. It may be a requirement but if it's on the roadmap then I'm still good with it.

Also, since I have your ear :) I'm exploring writing a small Student Information System (SIS) for schools that have small budgets. I've already got one (actually, it just does Immunization Tracking) in Silverlight that I'm converting to Blazor using Radzen. The Radzen version (Server Side Blazor) will be Version 2 and I'm looking to write Version 3 using Client Side Blazor. Version 3 will be the full blown SIS. Do you think Oqtane would be a good choice? I know you're not bias or anything :)

**Reply by ADefWebserver** [reply]
> The Radzen version (Server Side Blazor) will be Version 2 and I'm looking to write Version 3 using Client Side Blazor. Version 3 will be the full blown SIS. Do you think Oqtane would be a good choice?

@TrailBlazor - Wit Oqtane, you get the Server Side Blazor and the Client Side Blazor "at-the-same-time" because you simply configure Oqtane to run in either Client Side (WebAssembly) mode or Server Side Mode. Your module code would not change. Also, in my opinion Oqtane is a good choice for an SIS. Also, the point @sbwalker was trying to make, is that you could *still* use Radzen controls in your Oqtane module.

**Reply by TrailBlazor** [reply]
Hey Michael, thanks for chiming in. Yes, I've seen your most excellent blogs! I do wish there was better documentation on Oqtane. Maybe I just haven't looked hard enough :)

**Reply by TrailBlazor** [reply]
@ADefWebserver @sbwalker I think I'll just download it and play around a bit. I'm kind of fighting Radzen a bit after I've tweaked the MainLayout especially when I do browser reloads. Using Chris's SessionStorage library and that helps.

**Reply by ADefWebserver** [reply]
@TrailBlazor - These articles may help:

- [Blazor Oqtane Survey Module](https://blazorhelpwebsite.com/ViewBlogPost/48) - Uses Radzen in an Oqtane Module
- [Configuring The Blazor Oqtane Blog Module](https://blazorhelpwebsite.com/ViewBlogPost/15) - Shows how to configure Oqtane to run source code for a module (for example the Survey module)
